src/iccpd/src/iccp_netlink.c

```diff
@@ -0,0 +1,1461 @@
+    int len = 0;
+
+    sys = system_get_instance();
+    if (sys == NULL)
+        return 0;
+
+    genlmsg_parse(nlh, 0, attrs, TEAM_ATTR_MAX, NULL);
+
+    if (attrs[TEAM_ATTR_TEAM_IFINDEX])
+        ifindex = nla_get_u32(attrs[TEAM_ATTR_TEAM_IFINDEX]);
+
+    local_if = local_if_find_by_ifindex(ifindex);
+
+    if (!local_if)
+        return NL_SKIP;
+
+    if (local_if->type != IF_T_PORT_CHANNEL)
+        return NL_SKIP;
+
+    csm = local_if->csm;
+
+    if (csm)
+    {
+        if (!attrs[TEAM_ATTR_LIST_PORT])
+            return NL_SKIP;
+
+        nla_for_each_nested(nl_port, attrs[TEAM_ATTR_LIST_PORT], i)
+        {
+            uint32_t member_index;
+
+            if (nla_parse_nested(port_attrs, TEAM_ATTR_PORT_MAX, nl_port, NULL))
+            {
+                ICCPD_LOG_ERR(__FUNCTION__, "Failed to parse nested attributes.");
+                return NL_SKIP;
+            }
+
+            if (!port_attrs[TEAM_ATTR_PORT_IFINDEX])
+            {
+                ICCPD_LOG_ERR(__FUNCTION__, "ifindex port attribute not found.");
+                return NL_SKIP;
+            }
+
+            member_index = nla_get_u32(port_attrs[TEAM_ATTR_PORT_IFINDEX]);
+
+            local_if_member = local_if_find_by_ifindex(member_index);
+            if (local_if_member == NULL)
+            {
+
+                ICCPD_LOG_WARN(__FUNCTION__, "%s: Failed to find a port instance (%d).",
+                               local_if->name, member_index);
+                continue;
+            }
+
+            if (port_attrs[TEAM_ATTR_PORT_REMOVED] && local_if_member->po_id != -1)
+            {
+                local_if_member->po_id = -1;
+                mlacp_unbind_local_if(local_if_member);
+            }
+            else if ( local_if_member->po_id == -1)
+            {
+                local_if_member->po_id = local_if->po_id;
+                mlacp_bind_local_if(local_if->csm, local_if_member);
+            }
+        }
+
+        memset(temp_buf, 0, 512);
+        LIST_FOREACH(lif, &(MLACP(csm).lif_list), mlacp_next)
+        {
+            if (lif->type == IF_T_PORT && lif->po_id == local_if->po_id)
+            {
+                if (strlen(temp_buf) != 0)
+                    len += snprintf(temp_buf + len, 512 - len, "%s", ",");
+
+                len += snprintf(temp_buf + len, 512 - len, "%s", lif->name);
+            }
+        }
+
+        if (strcmp(temp_buf, local_if->portchannel_member_buf))
+        {
+            memset(local_if->portchannel_member_buf, 0, 512);
+            memcpy(local_if->portchannel_member_buf, temp_buf, sizeof(local_if->portchannel_member_buf) - 1);
+
+            if (MLACP(csm).current_state == MLACP_STATE_EXCHANGE)
+            {
+                /* portchannel member changed, update port isolate attribute*/
+                update_peerlink_isolate_from_all_csm_lif(csm);
+                csm->isolate_update_time = time(NULL);
+            }
+        }
+    }
+    else /*peerlink portchannel */
```

Comment:
Both port learn_mode setting and src_port field in port-isolation mode setting are done using port-channel name instead of its members' names, would you please explain why we need to retrieve the peer-link port-channel members from the kernel. In other words, should we trigger learn_mode and port-isolation settings directly from set_peer_link() using just port-channel?

Reply:
> Both port learn_mode setting and src_port field in port-isolation mode setting are done using port-channel name instead of its members' names, would you please explain why we need to retrieve the peer-link port-channel members from the kernel. In other words, should we trigger learn_mode and port-isolation settings directly from set_peer_link() using just port-channel?

@lisahnguyen, Please see the codes in function update_peerlink_isolate_from_all_csm_lif(), the src port is peer-link port or portchannel, the dst ports are all the members of portchannel.

Reply:
@jianjundong: since this section of the code handles peer-link which is always the source port (not dest ports) in port-isolation setting, we don't need the list of all member ports of a port-channel peer link. In the update_peerlink_isolate_from_all_csm_lif(), only dest ports are the list of port-channel's member ports

Reply:
@lisahnguyen, learn-mode is set on peer-link member port, so we need the list of all member ports of a port-channel peer link. You are correct, we don't need to call update_peerlink_isolate_from_all_csm_lif() when the member list of peer-link changed.

Reply:
@jianjungdong: There are two scenarios:
- Scenario 1:  an existing member is removed from a port-channel: in this case, we need to put back the learning mode on the actual port-channel member. Hence, set_peerlink_mlag_port_learn() should be called.  I agree with this change. 

- Scenario 2: a new member is added to an existing port channel: since set_peerlink_mlag_port_learn(),  sets learn mode on the port-channel (not the port-channel's members ports), is it correct that we should not call set_peerlink_mlag_port_learn() in this case?

Reply:
@lisahnguyen, We are not sure if a new member is added to an existing port channel, the learn mode of the new member follows the port channel in all vendor's implementation. So we call set_peerlink_mlag_port_learn() in this case, the parameter is port channel if, not member if. Thanks.